### DISCUSSION TITLE
Using .env file for our project is not viable

### DISCUSSION THREAD

**Discussion by Deivu** [prisma/prisma#18895] [Q&A] [answered]
We use a json file for our configuration, but I can't find any way of integrating the database url from a json file instead of a .env
Is there any ways to work around this?

I dont like the idea of having both .env and .json files for our configurations as that can complicate things

**Comment by nurul3101** [comment, marked as answer]
Hello @Deivu 👋 

One alternative I can think of is that you can programmatically override data source URL: [reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#programmatically-override-a-datasource-url)

You can read the database url from a JSON file and pass it the PrismaClient instance as shown in the reference example.

**Comment by janpio** [comment]
You don't have to use the `.env` file, that is just a shortcut to get the values loaded into the acutal environment on your server. You can do that in any way you choose. A shell script might work.
